src/include/sof/schedule/dp_schedule.h

```diff
@@ -86,4 +88,31 @@ int scheduler_dp_task_init(struct task **task,
 void scheduler_get_task_info_dp(struct scheduler_props *scheduler_props,
 				uint32_t *data_off_size);
 
+struct bind_info;
+struct sof_source;
+struct sof_sink;
+union scheduler_dp_thread_ipc_param {
```

Comment:
can we comment why this is a union ? i.e. when we use bind_data

Reply:
Event allows signaling multiple independent conditions and waiting until any or all of them are signaled. It also lets you choose which events to monitor. Using poll here seems unnecessary.